azext_iot/_params.py

```diff
@@ -557,7 +558,7 @@ def load_arguments(self, _):
             arg_type=get_enum_type(ProtocolType),
             help="Indicates device-to-cloud message protocol",
         )
-        context.argument("qos", arg_type=qos_type)
+        context.argument("qos", arg_type=qos_type, deprecate_info=context.deprecate(expiration='0.10.12'))
```

Comment:
Think we should take off the expiration because we aren't sure the exact release the overhauled device simulation is landing in.